core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/autoscales/AutoscaleV4Endpoint.java

```diff
@@ -152,6 +152,7 @@ void decommissionInternalInstancesForClusterCrn(@PathParam("crn") String cluster
             notes = Notes.STACK_NOTES, nickname = "autoscaleStopInstancesByCrn")
     void stopInstancesForClusterCrn(@PathParam("crn") String clusterCrn,
             @RequestBody @NotEmpty List<String> instanceIds,
+            @QueryParam("hostGroup") String hostGroup,
```

Comment:
Do we need to change the API here? Can be inferred from the instanceIds itself - there's a check further down the callchain to ensure that all instances belong to the same hostGroup.

Reply:
Since there was a requirement to add the hostGroup parameter in the corresponding CLI commands: https://github.infra.cloudera.com/thunderhead/thunderhead/pull/7193/files#diff-3de98e342d7cef2e66ff5a0e1a796b9fb70eb7d4587d787d2009d0b2967b4571R4109, I changed the API to accept this extra parameter as well. If this param is removed, then the `hostGroup` passed as input via the CLI will not be propagated to CB at all.

Reply:
The hostGroup will be added alongside an option to stop N instances, instead of specifying a list of instance ids. When a list is provided, the hostGroup can be figured out. Will the same API evolve to support that, or will that be a new one?

Reply:
The same API endpoint can be modified to support that operation. I think that would require further modification to accept the hostGroup and instanceCount as part of the request body along with instanceIds, as opposed to a query param.

Reply:
Is this a change we can make at a later point, when support is added to specify hostGroup+instanceCount.
2 usage scenarios: 
1. Specify instanceIds (supported today) - we can derive the hostGroup in this scenario.
2. Specify hostGroup + nodeCount (not supported today). This is where a hostGroup parameter will be needed. I'd prefer adding this parameter as part of supporting this specific scenario, whenever that change is made to support the scenario.

Reply:
Addressed this. Updated to remove the extra parameter.